libs/backend/src/ballot_package/ballot_package_io.test.ts

```diff
@@ -0,0 +1,197 @@
+import { fakeLogger } from '@votingworks/logging';
+import {
+  DEFAULT_SYSTEM_SETTINGS,
+  InsertedSmartCardAuth,
+} from '@votingworks/types';
+import {
+  fakeElectionManagerUser,
+  fakePollWorkerUser,
+} from '@votingworks/test-utils';
+import {
+  electionMinimalExhaustiveSampleFixtures,
+  electionFamousNames2021Fixtures,
+  systemSettings,
+} from '@votingworks/fixtures';
+import { assert } from '@votingworks/basics';
+import { safeParseSystemSettings } from '@votingworks/utils';
+import { createBallotPackageWithoutTemplates } from './test_utils';
+import { readBallotPackageFromUsb } from './ballot_package_io';
+import { createMockUsb } from '../mock_usb';
+
+test('readBallotPackageFromUsb can read a ballot package from usb', async () => {
+  const { electionDefinition } = electionMinimalExhaustiveSampleFixtures;
+  const authStatus: InsertedSmartCardAuth.AuthStatus = {
+    status: 'logged_in',
+    user: fakeElectionManagerUser({
+      electionHash: electionDefinition.electionHash,
+    }),
+  };
+
+  const mockUsb = createMockUsb();
+  mockUsb.insertUsbDrive({
+    'ballot-packages': {
+      'test-ballot-package.zip': createBallotPackageWithoutTemplates(
+        electionDefinition,
+        { systemSettingsString: systemSettings.asText() }
+      ),
+    },
+  });
+  const usbDrives = await mockUsb.mock.getUsbDrives();
+  const usbDrive = usbDrives[0];
+  assert(usbDrive);
+
+  const ballotPackageResult = await readBallotPackageFromUsb(
+    authStatus,
+    usbDrive,
+    fakeLogger()
+  );
+  assert(ballotPackageResult.isOk());
+  const ballotPackage = ballotPackageResult.ok();
+  expect(ballotPackage.electionDefinition).toEqual(electionDefinition);
+  expect(ballotPackage.systemSettings).toEqual(
+    safeParseSystemSettings(systemSettings.asText()).unsafeUnwrap()
+  );
+});
+
+test("readBallotPackageFromUsb uses default system settings when system settings don't exist in the zip file", async () => {
```

Comment:
Some of these tests are ~direct ports from https://github.com/votingworks/vxsuite/pull/3223/commits/ac709203fb5a83b492784ca318b98091d6a940a1#diff-ec10310b6ce5e9cd2721ee574ce861f62cab75215f744ae2b105c716492b3583L130

But there are also a few net new ones. Test descriptions should match up pretty closely